### PR TITLE
refactor(theme-classic): clean up CSS of doc sidebar item

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

After a cursory look of the final source code produced by Docusaurus, I found that there are two redundant CSS classes for doc sidebar items. So it's safe to remove these classes and the related CSS code.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Preview

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/DocSidebarItem/styles.module.css [hunk 1]
```diff
@@ -6,18 +6,6 @@
  */
 
 @media (min-width: 997px) {
-  .menuLinkText {
-    cursor: initial;
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/DocSidebarItem/styles.module.css:10]: If the `href` attribute is missing in a link, the default cursor is used, so there is no need for this class.

File: packages/docusaurus-theme-classic/src/theme/DocSidebarItem/styles.module.css [hunk 2]
```diff
@@ -6,18 +6,6 @@
...
-  }
-
-  .menuLinkText:hover {
-    background: none;
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/DocSidebarItem/styles.module.css:14]: Also, we don't need this CSS class after introducing `menu__list-item-collapsible`.

File: website/sidebars.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ const sidebars = {
         type: 'generated-index',
       },
       collapsed: false,
+      collapsible: false,
```
Comment (lex111) [inline website/sidebars.js:21]: Let's disable collapsing for this category as a dogfooding approach.

Comment (slorber) [reply website/sidebars.js:18]: We have another section to dogfood now, maybe you can move these tests there? 

(one with/without link and with/without collapsible?)

https://deploy-preview-6622--docusaurus-2.netlify.app/tests/docs

**Review by slorber — APPROVED** [review]
welcome back 😄 

LGTM, just move dogfood to appropriate section so that we can review more edge-cases

(I'm particularly interested to inspect the site with JS disable in multiple scenarios because this is why we added `hrefWithSSRFallback`)

File: website/sidebars.js (hunk 3 above)
Comment (lex111) [reply website/sidebars.js]: Oh yes, that's right, I forgot about that. I guess it's okay if I disable collapsing for the category "HTML items tests".

**Comment by lex111** [comment]
Thanks! I still want to implement some of my ideas in the near future.

File: website/sidebars.js [hunk 4]
```diff
@@ -14,10 +14,6 @@ const sidebars = {
     {
       type: 'category',
       label: 'Getting Started',
-      link: {
-        type: 'generated-index',
-      },
-      collapsed: false,
```
Comment (Josh-Cena) [inline website/sidebars.js:20]: This should be restored

Comment (lex111) [reply website/sidebars.js]: :ok_hand:

**Comment by lex111** [comment]
It's make sense to get a full picture, although it is not necessary for this fix.
Just added a new category right after main category.

**Comment by Josh-Cena** [comment]
The effect looks good to me!
